api/saml/provider.py

```diff
@@ -74,6 +74,21 @@ def __init__(self, library, integration, analytics=None):
         self._configuration_factory = SAMLConfigurationFactory(SAMLMetadataParser())
         self._authentication_manager_factory = SAMLAuthenticationManagerFactory()
 
+        db = Session.object_session(library)
+
+        with self.get_configuration(db) as configuration:
+            self._patron_id_use_name_id = (
+                str(configuration.patron_id_use_name_id) == "1"
```

Comment:
This is super ugly and can be fixed in two different ways:
1. Adding a new method to `ConfigurationMetadata` class:
```python
class ConfigurationMetadata(object):
    @property
    def bool_value(self):
         return str(self) == "1"
```
And replacing the code above with the following:
```python
self._patron_id_use_name_id = configuration.patron_id_use_name_id.bool_value
```

2. Changing the type of `patron_id_use_name_id` from the integer text input to a dropdown, adding the following method to `ConfigurationMetadata` class:
```python
class ConfigurationMetadata(object):
    @property
    def bool_value(self):
         return str(self).lower() in ConfigurationSetting.MEANS_YES
```
And replacing the code above with the following:
```python
self._patron_id_use_name_id = configuration.patron_id_use_name_id.bool_value
```

Reply:
I prefer the second solution due to its similarity to ConfigurationSetting.bool_value.

Reply:
As I described in [server_core/PR # 1253](https://github.com/NYPL-Simplified/server_core/pull/1253) I couldn't make it a property because `ConfigurationMetadata` has `__get__` method.